supervision/metrics/detection.py

```diff
@@ -456,3 +462,373 @@ def plot(
+        EPS=1e-16,
+    ):
+        """Compute the average precision, given the recall and precision curves.
+        Source: https://github.com/rafaelpadilla/Object-Detection-Metrics.
+        # Arguments
+            matches:  True positives (nparray, nx1 or nx10).
+            prediction_confidence:  Objectness value from 0-1 (nparray).
+            prediction_class_ids:  Predicted object classes (nparray).
+            true_batch_class_ids:  True object classes (nparray).
+        """
+        sorted_confidences = np.argsort(-prediction_confidence)
+        matches = matches[sorted_confidences]
+        prediction_class_ids = prediction_class_ids[sorted_confidences]
+
+        # Find unique classes
+        unique_classes, class_counts = np.unique(
+            true_batch_class_ids, return_counts=True
+        )
+        num_classes = unique_classes.shape[0]  # number of classes, number of detections
+
+        average_precisions = np.zeros((num_classes, matches.shape[1]))
+        for ci, c in enumerate(unique_classes):
+            valid = prediction_class_ids == c
+            num_targets = class_counts[ci]  # number of labels
+            num_predictions = valid.sum()  # number of predictions
+            if num_predictions == 0 or num_targets == 0:
+                continue
+
+            fp_pool = (1 - matches[valid]).cumsum(0)
+            tp_pool = matches[valid].cumsum(0)
+
+            recall = tp_pool / (num_targets + EPS)
+            precision = tp_pool / (tp_pool + fp_pool)
+
+            for j in range(matches.shape[1]):
+                average_precisions[
+                    ci, j
+                ] = MeanAveragePrecision.compute_average_precision(
+                    recall[:, j], precision[:, j]
+                )
+
+        return average_precisions
+
+    @classmethod
+    def _validate_input_tensors(
+        cls, predictions: List[np.ndarray], targets: List[np.ndarray]
+    ):
+        """
+        Checks for shape consistency of input tensors.
+        """
+        if len(predictions) != len(targets):
+            raise ValueError(
+                f"Number of predictions ({len(predictions)}) and targets ({len(targets)}) must be equal."
+            )
+        if len(predictions) > 0:
+            if not isinstance(predictions[0], np.ndarray) or not isinstance(
+                targets[0], np.ndarray
+            ):
+                raise ValueError(
+                    f"Predictions and targets must be lists of numpy arrays. Got {type(predictions[0])} and {type(targets[0])} instead."
+                )
+            if predictions[0].shape[1] != 6:
+                raise ValueError(
+                    f"Predictions must have shape (N, 6). Got {predictions[0].shape} instead."
+                )
+            if targets[0].shape[1] != 5:
+                raise ValueError(
+                    f"Targets must have shape (N, 5). Got {targets[0].shape} instead."
+                )
+
+    def to_dict(self):
```

Comment:
`MeanAveragePrecision` is `dataclass`. You don't need to implement `to_dict`. It is already implemented by `dataclasses` package. Take a look [here](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict). I'd drop it.